src/hb-ot-name-table.hh

```diff
@@ -200,7 +223,19 @@ struct name
 
     const void *dst_string_pool = &(this + this->stringOffset);
 
-    c->copy_all (it, src_string_pool, dst_string_pool);
+    NameRecord *name_records = (NameRecord *) calloc (it.len (), NameRecord::static_size);
+    hb_array_t<NameRecord> records (name_records, it.len ());
```

Comment:
Use hb_vector_t and call alloc () to size it and push() to add. Then you don't need to do the calloc()/free() and memcpy(). After that you can qsort() and pass .iter() to the copy_all as you're currently doing.

Reply:
Oh actually, you don't need the temporary array/vector I think you should be able to just qsort() on nameRecordZ after you copy_all too it.

Reply:
> Oh actually, you don't need the temporary array/vector I think you should be able to just qsort() on nameRecordZ after you copy_all too it.

Right.  That's what I thought as well.

Reply:
the resolve_links () is called after serialize (), if we qsort after copy_all, the offset value might be set to wrong value.

Also the NameRecord doesn't overload operator = function, so the push() in vector doesn't work. According to coding style, we don't usually overload "=" for struct?

Reply:
Ah okay, makes sense. Then this looks good as is.